src/contributing/prs.md

```diff
@@ -0,0 +1,150 @@
+# Pull Requests
+
+We welcome all and any contributions, from code to documentation to examples,
+submitted as *pull requests* to one of our [GitHub
+repositories](https://github.com/ykjit/). Pull requests are reviewed by a yk
+contributor: changes may be requested in an ongoing dialogue between reviewee
+and reviewer; and, once agreement is reached, the pull request will be merged.
+
+This page explains yk's approach to the pull request model, based on
+observations of good practise in other projects, and our own experiences.
+Please note that this is a living process: the details are subject to continual
+refinement and change, and no document can perfectly capture all the
+possibilities that will crop up in real life.
+
+
+## What makes a good pull request?
+
+The typical aim of a pull request to a yk repository is to make a focussed
+change such as: add a new feature; fix a bug; or edit documentation. Good pull
+requests are:
+
+  1. *Self-contained*. A pull request should not try and change everything at
+     once. Ideally a pull request fixes a single bug or adds a single feature;
+     but sometimes a set of interdependent fixes or features makes more sense.
+     In general, small pull requests are better than big pull requests.
+
+  2. *Tested*. Every piece of new functionality must have at least one
+     accompanying test. Authors should also strive not to break any other part
+     of the system. yk runs various tests before a pull request is merged: this
+     minimises, but does not remove, the chances of a merged pull request
+     breaking the repository: careful thought and testing on the part of the
+     pull request author are still required.
+
+  3. *Documented*. Code diffs show *what* has been changed, but not *why*.
+     Documentation explains to other humans the context for a change (i.e. why
+     something needed to be changed), the reason why the change is as it is
+     (i.e. could the change have taken a different form? what alternatives were
+     considered or attempted?), and the consequences of doing so (i.e. does
+     this make something easier or harder in the future?). Documentation comes
+     in the form of comments in code, “external” documentation (such as you
+     read here), and commit and pull request messages. In whatever form it
+     comes, documentation must be clear (as easy to understand as possible),
+     concise (as short as possible while getting all the required points
+     across), and complete (not missing any important points).
```

Comment:
Can we add a note about not adding unnecessary documentation? Over time, I've come to the feeling that over-documenting invites stale documentation.

Reply:
Fixed in 21c7e7d.